### PR TITLE
Add BitConverter.ToUInt16 / BitConverter.ToUInt32 polyfill

### PR DESCRIPTION
**Pull request by MichelZ**
## Why
`BitConverter.ToUInt16(ReadOnlySpan<Byte>)` and `BitConverter.ToUInt32(ReadOnlySpan<Byte>)` are available on newer TFMs but missing from the current polyfill surface. Adding them keeps `ReadOnlySpan<Byte>`-based BitConverter APIs consistent with the existing `ToInt16` and `ToInt32` polyfills.

## What changed
- Added new polyfills for:
  - `System.BitConverter.ToUInt16(ReadOnlySpan<Byte>)`
  - `System.BitConverter.ToUInt32(ReadOnlySpan<Byte>)`
- Implementation follows existing project patterns:
  - Delegates to the existing `BitConverter.ToUInt16(byte[], int)` / `BitConverter.ToUInt32(byte[], int)` array-based overload via `value.ToArray()`
- Added test coverage in SystemTests for:
  - Representative values (42, 256)
  - Boundary values (`ushort.MinValue`, `ushort.MaxValue`, `uint.MinValue`, `uint.MaxValue`)
- Regenerated metadata/docs outputs and bumped package version patch:
  - `polyfill-supported-versions.json`
  - `README.md`
  - `Meziantou.Polyfill.csproj` version 1.0.116 → 1.0.117

## Notes
This is intentionally a straightforward implementation matching the existing `BitConverter.ToInt16(ReadOnlySpan<Byte>)`  / `.BitConverter.ToInt32(ReadOnlySpan<Byte>)` polyfill style in this repository.

### CODE REVIEW

**Review by meziantou — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ The filtering logic works as follows:
 - `System.ValueTuple<T1, T2, T3, T4, T5, T6, T7>`
 - `System.ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> where TRest : struct`
 
-### Methods (538)
+### Methods (539)
```
Comment (meziantou) [inline README.md]:
```suggestion
### Methods (540)
```
